### PR TITLE
Dataframes will now replaceOne with upsert if they contain an _id.

### PR DESCRIPTION
**Pull request by rozza**
SPARK-66 SPARK-74

### CODE REVIEW

File: src/main/scala/com/mongodb/spark/MongoSpark.scala [hunk 1]
```diff
@@ -137,6 +138,77 @@ object MongoSpark {
...
+   * '''Note:''' If the dataFrame contains an `_id` field the data will upserted and replace any existing documents in the collection.
+   *
+   * @param dataFrame the DataFrame to save to MongoDB
+   * @param writeConfig the writeConfig
+   * @since 1.1.0
+   */
+  def save(dataFrame: DataFrame, writeConfig: WriteConfig): Unit = {
+    val mongoConnector = MongoConnector(writeConfig.asOptions)
+    val documentRdd: RDD[BsonDocument] = dataFrame.rdd.map(row => rowToDocument(row))
+    dataFrame.schema.fields.exists(_.name == "_id") match {
```
Comment (evanchooly) [inline src/main/scala/com/mongodb/spark/MongoSpark.scala]: wouldn't if/else be simpler?

**Comment by evanchooly** [comment]
+1

File: src/main/scala/com/mongodb/spark/MongoSpark.scala (hunk 1 above)
Comment (rozza) [inline src/main/scala/com/mongodb/spark/MongoSpark.scala]: Good point 👍

**Comment by rozza** [comment]
Rebased and merged

**Comment by FS1360472174** [comment]
why not support rdd

**Comment by sekys** [comment]
please support RDD ...

**Comment by rozza** [comment]
> please support RDD ...

Please add a ticket to https://jira.mongodb.org/browse/SPARK - the reason there is no current RDD support is due to RDD's containing any serializable type, that could be converted to BSON by the codec registry.

**Pull request closed without merging**